File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.php
class Solution {

    /**
     * @param String $haystack
     * @param String $needle
     * @return Integer
     */
    function strStr($haystack, $needle) {
        $lh = strlen($haystack);
        $ln = strlen($needle);
        
        for($i = 0; $i < ($lh - $ln + 1); $i++){
            if(substr($haystack, $i, $ln) == $needle){
                return $i;
            }
        }
        
        return -1;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    n := len(needle)
    
    if n == 0 {
        return n
    }
    
    for i := 0; i < len(haystack) - n + 1; i++ {
        if haystack[i:i+n] == needle {
            return i
        }
    }
    
    return -1
}